src/object_store/wrapped.rs

```diff
@@ -89,16 +85,15 @@ impl InternalObjectStore {
     // Get the table prefix relative to the root of the internal object store.
     // This is either just a UUID, or potentially UUID prepended by some path.
     pub fn table_prefix(&self, table_prefix: &str) -> Path {
-        match self.config.clone() {
-            schema::ObjectStore::S3(S3 {
-                prefix: Some(prefix),
-                ..
-            })
-            | schema::ObjectStore::GCS(GCS {
-                prefix: Some(prefix),
-                ..
-            }) => Path::from(format!("{prefix}/{table_prefix}")),
-            _ => Path::from(table_prefix),
+        let prefix = match self.config.clone() {
+            ObjectStoreConfig::AmazonS3(aws_config) => aws_config.prefix,
+            ObjectStoreConfig::GoogleCloudStorage(google_config) => google_config.prefix,
+            _ => None,
+        };
+
+        match prefix {
+            Some(prefix) => Path::from(format!("{prefix}/{table_prefix}")),
+            None => Path::from(table_prefix),
         }
```

Comment:
Seems simpler to do it with one pattern match as before?

```suggestion
        match self.config.clone() {
            ObjectStoreConfig::AmazonS3(S3Config {
                prefix: Some(prefix),
                ..
            })
            | ObjectStoreConfig::GoogleCloudStorage(GCSConfig {
                prefix: Some(prefix),
                ..
            }) => Path::from(format!("{prefix}/{table_prefix}")),
            _ => Path::from(table_prefix),
        }
```

Reply:
I reverted, though I think the readability was better with the change.